packages/graphile-utils/src/makeExtendSchemaPlugin.ts

```diff
@@ -118,11 +119,26 @@ export default function makeExtendSchemaPlugin(
         typeof generator === "function"
           ? generator(build, schemaOptions)
           : generator;
-      if (!(typeDefs as any) || (typeDefs as any).kind !== "Document") {
-        throw new Error(
-          "The first argument to makeExtendSchemaPlugin must be generated by the `gql` helper"
-        );
+      if (Array.isArray(typeDefs)) {
+        if (typeDefs.some(typeDef => (typeDef as any).kind !== "Document")) {
+          throw new Error(
+            "All elements given to an array passed as the first argument of makeExtendSchemaPlugin must be generated by the `gql` helper."
+          );
+        }
+      } else {
+        if (!(typeDefs as any) || (typeDefs as any).kind !== "Document") {
+          throw new Error(
+            "The first argument to makeExtendSchemaPlugin must be generated by the `gql` helper, or an array of the same."
+          );
+        }
       }
+
+      const typeDefsArr = Array.isArray(typeDefs) ? typeDefs : [typeDefs];
+      const mergedTypeDefinitions = typeDefsArr.reduce(
+        (definitions, typeDef) => definitions.concat(typeDef.definitions),
+        [] as DefinitionNode[]
+      );
```

Comment:
Please can you move the assertions above into the reducer?

```ts
      const typeDefsArr = Array.isArray(typeDefs) ? typeDefs : [typeDefs];
      const mergedTypeDefinitions = typeDefsArr.reduce(
        (definitions, typeDef) => {
          if (!(typeDef as any) || (typeDef as any).kind !== "Document") {
            throw new Error(
              "The first argument to makeExtendSchemaPlugin must be generated by the `gql` helper, or an array of the same."
            );
          }
          definitions.push(...typeDef.definitions);
          return definitions;
        },
        [] as DefinitionNode[]
      );
```

That way we only have to loop once, and we only throw the error in one place. I've also changed `concat` to `push(...` for [performance reasons](https://jsperf.com/javascript-array-concat-vs-push/294).